operator/operator.php

```diff
@@ -64,12 +74,36 @@ public function get_user_categories()
 	 */
 	public function set_user_categories($forum_id)
 	{
-		// Set the collapsed category data array
-		$this->set_collapsed_categories($forum_id);
+		// Do not apply the ext functionality for bots
+		if (!$this->user->data['is_bot'])
+		{
+			// Set the collapsed category data array
+			$this->set_collapsed_categories($forum_id);
+
+			// Update the db with json encoded array of collapsed category data ($this->collapsed_categories)
+			if ($this->user->data['is_registered'])
+			{
+				$sql = 'UPDATE ' . USERS_TABLE . "
+					SET collapsible_categories = '" . $this->db->sql_escape(json_encode($this->collapsed_categories)) . "'
+					WHERE user_id = " . (int) $this->user->data['user_id'];
+				$this->db->sql_query($sql);
+			}
+
+			// If there was an error updating the user's data, return false
+			// Otherwise update cookie data as well
+			if (!$this->db->sql_affectedrows())
+			{
+				return false;
+			}
+			else
+			{
+				$this->set_cookie_categories($forum_id);
```

Comment:
This should not be in an else. This should always run. We want users to have both their db and cookie updated.

It should be where your `return true` is, but like::
```
// Set a cookie with the collapsed forum data
return $this->set_cookie_categories($forum_id);
```